geniza/footnotes/tests/test_footnote_models.py

```diff
@@ -45,32 +51,104 @@ def test_str(self, source, twoauthor_source, multiauthor_untitledsource):
         ]
         assert str(multiauthor_untitledsource) == "%s, %s, %s and %s" % tuple(lastnames)
 
-    @pytest.mark.django_db
-    def test_str_article(self, article):
-
-        # article with title, journal title, volume, year
-        assert str(article) == '%s, "%s" %s %s (%s)' % (
-            article.authors.first().firstname_lastname(),
-            article.title,
-            article.journal,
-            article.volume,
-            article.year,
-        )
-        # article with no title
-        article.title = ""
-        assert str(article) == "%s, %s %s (%s)" % (
-            article.authors.first().firstname_lastname(),
-            article.journal,
-            article.volume,
-            article.year,
-        )
-        # no volume
-        article.volume = ""
-        assert str(article) == "%s, %s (%s)" % (
-            article.authors.first().firstname_lastname(),
-            article.journal,
-            article.year,
-        )
+        #                                            #
```

Comment:
This looks like you're duplicating a lot of the logic in the code you're trying to test, which isn't a good approach.

We have several pyest source fixtures in `geniza.footnotes.conftest` — you should test against them, similar to the way the previous `test_str_article` worked. You include the fixture name as an argument to your test method, and pytest will create the fixture so you can test against it. I think having methods for each type would make sense, and you can add new fixtures for types we don't already have represented.